src/common/http-request/join-or-merge-includes.ts

```diff
@@ -0,0 +1,19 @@
+import { CheckoutIncludes, CheckoutIncludeParam } from '../../checkout';
+
+import joinIncludes from './join-includes';
+import mergeIncludes from './merge-includes';
+
+/**
+ * Joins or merges a base list of includes with a set of additional includes.
+ */
+export default function joinOrMergeIncludes(
+    baseIncludes: string[],
+    includeDictionaryOrList: CheckoutIncludeParam | CheckoutIncludes[] = []
```

Comment:
I think `includeDictionaryOrList` needs to be a generic type instead of `CheckoutIncludeParam | CheckoutIncludes[]`, as this function belongs to the `common` so ideally it should not be tied to a specific set of params.